### PR TITLE
Fix KSP properties interception

### LINKED ISSUE #9579: @RequestScope no longer works in micronaut 4.0
### Expected Behavior

@RequestScope continues to work like in micronaut 3.9. The following two features no longer work:
- Default value assignments in Kotlin class constructors of @RequestScope classes
- @RequestScope objects should be freshly initialized for each http request.

### Actual Behaviour

- Default value assignments in Kotlin class constructors are ignored. The value is assigned null. If the type is defined as non-nullable the request fails.
- @RequestScope object behaves like a Singleton and retains state of previous requests.

### Steps To Reproduce

- Assign a default value to a @RequestScope object, e.g:
```
@RequestScope
class RequestScopeClass(
    // this no longer works in micronaut 4.0, default value is ignored and as a result type can't be non-null
    var text: String = "defaultText",
)
```

- Assign a request scoped value and check in a subsequent request if the assigned value was reset.

The example application contains tests for both cases.

### Environment Information

- Linux
- JDK 17

### Example Application

https://github.com/StefanSrf/micronaut-4-0-request-scope-bug

### Version

4.0.0

### PR DESCRIPTION
**Pull request by dstepanov**
Fixes https://github.com/micronaut-projects/micronaut-core/issues/9579

### CODE REVIEW

**Review by sdelamo — COMMENTED** [review]
There is a test failure https://ge.micronaut.io/s/ykb6cjysjyamc/tests/task/:inject-kotlin:test/details/io.micronaut.kotlin.processing.aop.compile.RequestBeanScope/test%20request%20scope%20bean%20with%20properties?top-execution=1

**Review by yawkat — APPROVED** [review]
amazing that removing code was all that was necessary to fix the bug

File: test-suite-kotlin-ksp/src/test/kotlin/io/micronaut/docs/server/request_scope/TestControllerTest.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package io.micronaut.docs.server.request_scope
+
+import io.micronaut.http.client.HttpClient
+import io.micronaut.http.client.annotation.Client
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.RepeatedTest
+import org.junit.jupiter.api.Test
+
+@MicronautTest
```
Comment (sdelamo) [inline test-suite-kotlin-ksp/src/test/kotlin/io/micronaut/docs/server/request_scope/TestControllerTest.kt:12]: Annotate with `@Property(name = "spec.name", value = "TestControllerTest")`

File: test-suite-kotlin-ksp/src/test/kotlin/io/micronaut/docs/server/request_scope/DemoController.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package io.micronaut.docs.server.request_scope
+
+import io.micronaut.http.HttpResponse
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.annotation.QueryValue
+
+@Controller
```
Comment (sdelamo) [inline test-suite-kotlin-ksp/src/test/kotlin/io/micronaut/docs/server/request_scope/DemoController.kt:10]: Annotate with `@Requires(property = "spec.name", value = "TestControllerTest")`

**Review by sdelamo — CHANGES_REQUESTED** [review]
Please use `spec.name` to avoid bean pollution

**Review by sdelamo — APPROVED** [review]

**Pull request merged**